rasa/core/validator.py

```diff
@@ -32,124 +32,142 @@ def __init__(self, domain: Domain, intents: TrainingData, stories: List[StorySte
 
         Verifies if the intents used in the stories are valid, and whether
         all valid intents are used in the stories."""
 
-        domain_intents = self.verify_intents()
+        everything_is_alright = self.verify_intents()
 
-        stories_intents = set()
-        for story in self.stories:
-            for event in story.events:
-                if type(event) == UserUttered:
-                    intent = event.intent["name"]
-                    stories_intents.add(intent)
-                    if intent not in domain_intents:
-                        logger.error(
-                            "The intent '{}' is used in stories, but is not "
-                            "listed in the domain file.".format(intent)
-                        )
-
-        for intent in domain_intents:
+        stories_intents = {
+            event.intent["name"]
+            for story in self.stories
+            for event in story.events
+            if type(event) == UserUttered
+        }
+
+        for story_intent in stories_intents:
+            if story_intent not in self.domain.intents:
+                logger.error(
+                    "The intent '{}' is used in stories, but is not "
+                    "listed in the domain file.".format(story_intent)
+                )
+                everything_is_alright = False
+
+        for intent in self.domain.intents:
             if intent not in stories_intents:
                 logger.warning(
                     "The intent '{}' is not used in any story.".format(intent)
                 )
+                everything_is_alright = ignore_warnings and everything_is_alright
+
+        return everything_is_alright
 
-    def verify_utterances(self):
+    def _gather_utterance_actions(self) -> Set[Text]:
+        """Return all utterances which are actions."""
+        return {
+            utterance
+            for utterance in self.domain.templates.keys()
+            if utterance in self.domain.action_names
+        }
+
+    def verify_utterances(self, ignore_warnings: bool = True) -> bool:
         """Compares list of utterances in actions with utterances in templates."""
 
         actions = self.domain.action_names
-        utterance_templates = set()
-        valid_utterances = set()
-
-        for utterance in self.domain.templates:
-            utterance_templates.add(utterance)
+        utterance_templates = set(self.domain.templates)
+        everything_is_alright = True
 
         for utterance in utterance_templates:
-            if utterance in actions:
-                valid_utterances.add(utterance)
-            else:
-                logger.error(
+            if utterance not in actions:
+                logger.warning(
                     "The utterance '{}' is not listed under 'actions' in the "
-                    "domain file.".format(utterance)
+                    "domain file. It can only be used as a template.".format(utterance)
```

Comment:
What does this mean? Utterances don't actually need to be listed in actions to be used?

Reply:
no they don't. you can also just use `dispatcher.utter_template(...)` from a custom action to use a defined template